meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc

```diff
@@ -30,8 +37,11 @@ COMPATIBLE_HOST_aarch64 = 'null'
 
 EFI_BOOT_PATH = "/boot/efi/EFI/BOOT"
 
-GRUB_BUILDIN_append += " chain ${@'efivar mok2verify password_pbkdf2' \
-                                  if d.getVar('UEFI_SB', True) == '1' else ''}"
+GRUB_SIGNING_MODULES += "${@'pgp gcry_rsa gcry_sha256 gcry_sha512 --pubkey %s ' % d.getVar('GRUB_PUB_KEY', True) if d.getVar('GRUB_SIGN_VERIFY', True) == '1' else ''}"
+
+GRUB_BUILDIN_append += "tftp reboot ${GRUB_SIGNING_MODULES} chain \
+  ${@'efivar password_pbkdf2 ' if d.getVar('UEFI_SB', True) == '1' else ''} \
+  ${@'mok2verify ' if d.getVar('UEFI_SELOADER', True) == '1' else ''}"
```

Comment:
We're losing some readability here by not splitting GRUB_SIGNING_MODULES to multiple lines still.  Also, can we do things with a bit less inline python?  Perhaps one variable for mok2verify, one for the new grub key method and then 3 appends to GRUB_BUILDIN, one for what they both need and another for each?

Reply:
Are you advocating something like follows?   The expression syntax of the recipe leaves a lot to be desired, but it is the only way to have it work properly and have conditional functionality. 

GRUB_SECURE_BOOT_MODULES += "${@'efivar password_pbkdf2 ' if d.getVar('UEFI_SB', True) == '1' else ''}"

GRUB_SIGNING_MODULES += "${@'pgp gcry_rsa gcry_sha256 gcry_sha512 --pubkey %s ' % d.getVar('GRUB_PUB_KEY', True) \
  if d.getVar('GRUB_SIGN_VERIFY', True) == '1' else ''}"

GRUB_SELOADER_MODULES += "${@'mok2verify ' if d.getVar('UEFI_SELOADER', True) == '1' else ''}"

GRUB_BUILDIN_append += "tftp reboot chain \
  ${GRUB_SECURE_BOOT_MODULES} \
  ${GRUB_SIGNING_MODULES} \
  ${GRUB_SELOADER_MODULES}"

Reply:
Something like that, yes please, thanks!